src/app/core/store/navigation/navigation.state.ts

```diff
@@ -51,3 +51,4 @@ export const selectWorkspace = createSelector(selectNavigation, (state: Navigati
 export const selectSearchTab = createSelector(selectNavigation, (state: NavigationState) => state.searchTab);
 export const selectUrl = createSelector(selectNavigation, state => state.url);
 export const selectPreviousUrl = createSelector(selectNavigation, state => state.previousUrl);
+export const selectViewCode = createSelector(selectWorkspace, workspace => workspace && workspace.viewCode || null);
```

Comment:
There is no need to specify `null` here since `undefined` will be returned if either `workspace` or `workspace.viewCode` is `undefined`.